### PR TITLE
Print type inference messages, plus the beginnings of static analysis

### PR DESCRIPTION
**Pull request by hinescd**

### CODE REVIEW

File: src/main/java/com/github/lessjava/LJCompiler.java [hunk 1]
```diff
@@ -111,29 +102,58 @@ public static void main(String[] args) throws IOException {
...
+
         // Compile the generated Java source
         JCompiler.compile();
 
         // TODO: exec junit
     }
 
+    private static void parseArgs(String[] args) {
+        if (args.length == 0) {
+            System.err.println("usage: LJCompiler [OPTIONS] <File>");
```
Comment (Zamua) [inline src/main/java/com/github/lessjava/LJCompiler.java]: We should consider using docopt: https://github.com/docopt/docopt.java

It’s awesome

File: src/main/java/com/github/lessjava/types/ast/ASTNode.java [hunk 2]
```diff
@@ -58,6 +58,8 @@
 public abstract class ASTNode {
     public Map<String, Object> attributes;
 
+    public int lineNo;
```
Comment (Zamua) [inline src/main/java/com/github/lessjava/types/ast/ASTNode.java]: Annoying, but I see no reason to abbreviate this. “lineNumber” might be more clear. Less likely to forget what it is later and not all that much verbose

File: src/main/java/com/github/lessjava/visitor/impl/LJStaticAnalysis.java [hunk 3]
```diff
@@ -83,6 +91,9 @@ public void preVisit(ASTConditional node) {
     @Override
     public void postVisit(ASTConditional node) {
         super.postVisit(node);
+        if(!node.condition.type.equals(new HMTypeBase(HMType.BaseDataType.BOOL))) {
+            addError(node.condition.type + " is not a boolean expression on line " + node.lineNo);
```
Comment (Zamua) [inline src/main/java/com/github/lessjava/visitor/impl/LJStaticAnalysis.java]: The line number should be prepended I think. Either in English “on line x, this error happened”

Or with some kind of separator “line x: this error happened”

I think including the line number should also be the job of the addError method, not something repeated in every error string

**Comment by hinescd** [comment]
Addressed requested changes, aside from docopt

**Pull request merged**